setup.py

```diff
@@ -429,12 +472,25 @@ def setup_common_extension() -> CMakeExtension:
     Also builds JAX or userbuffers support if needed.
 
     """
-    cmake_flags = []
-
+    # FindPythonInterp and FindPythonLibs are deprecated in newer CMake versions,
+    # but PyBind11 still tries to use them unless we set PYBIND11_FINDPYTHON=ON.
+    cmake_flags = [ "-DPYBIND11_FINDPYTHON=ON" ]
+
+    # Optionally switch userbuffers bootstrapping to the old MPI method.
+    # NOTE: This requires launching PyTorch distributed runs with
+    #       `mpiexec -np <N> -x MASTER_ADDR=<host addr> -x MASTER_PORT=<host port> -x PATH ...`
+    #       instead of `torchrun --nproc-per-node=<N> ...`
     if int(os.getenv("UB_MPI_BOOTSTRAP", "0")):
         assert os.getenv("MPI_HOME"), \
             "MPI_HOME must be set if UB_MPI_BOOTSTRAP=1"
-        cmake_flags.append("-DUB_MPI_BOOTSTRAP")
+        cmake_flags += [ "-DUB_MPI_BOOTSTRAP=ON" ]
+
+    # If we need to build TE/PyTorch extensions later, we need to compile core TE library with
+    # the same C++ ABI version as PyTorch.
+    if "pytorch" in frameworks():
+        import torch
+        if "-D_GLIBCXX_USE_CXX11_ABI=1" in torch._C._show_config():  # pylint: disable=unsupported-membership-test
+            cmake_flags += [ "-DUSE_CXX11_ABI=ON" ]
```

Comment:
We should make this independent of the default value for `USE_CXX11_ABI`:
```suggestion
        cmake_flags.append(
            f"-DUSE_CXX11_ABI={int(torch.compiled_with_cxx11_abi())}"
        )
```